### ISSUE TITLE
Lambda in VPC

### ISSUE THREAD

**Issue by ScOut3R** [aws/aws-xray-daemon#26]
If I understand correctly the X-Ray daemon is provisioned automatically for Lambda function.

Is this true for VPC based Lambdas?

I am having the following issue with a VPC based Lambda which uses X-Ray:

```
Error: connect ECONNREFUSED 169.254.79.2:2000
    at Object._errnoException (util.js:1022:11)
    at _exceptionWithHostPort (util.js:1044:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)
```

**Comment by awssandra** [comment]
Hi ScOut3R,

Sorry for the late reply. Let me take a look into this. Thanks!

**Comment by mattsoftware** [comment]
Any further issue on this? I am expecting to see traces from my lambda function. I am only new at xray but immediately saw traces with other functions. Running this one in vpc seems to not generate any.

**Comment by mattsoftware** [comment]
Sorry, user error... I had done everything but tick the 'enable xray' button on the lambda function.

**Comment by ScOut3R** [comment]
@srprash Could you pleas reopen? The OP issue is still valid and I have X-Ray enabled on the Lambda. :) In fact I receive the tracing information, but that error messes up the logs and the Lambda execution is marked as a failure.

**Comment by srprash** [comment]
Hi @ScOut3R Sorry for closing this one without resolution. I'm reopening the issue and will investigate. Thanks!

**Comment by srprash** [comment]
Hi @ScOut3R ,
I tested a lambda function in a VPC with a public subnet and a security group with open inbound and outbound rules. I was able to see the traces generated for my lambda function. With this I can say that X-ray would work for lambdas in VPC too. 
I would like to know what is your configuration of the VPC, subnet and the security group  for the lambda since the issue could be due to insufficient permissions.
